src/erl_optics.erl

```diff
@@ -339,3 +346,12 @@ get_lens(Key) ->
 
 get_optics() ->
     foil:lookup(?NS, optics).
+
+check_ets(Key) ->
+    case ets:lookup(erl_optics_ets, Key) of
```

Comment:
There's a possible race condition between the lookup and the insert. You can use `update_counter` to make this an atomic operation.

```erlang
1> Tid = ets:new(test, []).
#Ref<0.1069691747.2281832452.174368>
2> ets:update_counter(Tid, test, 1, {test, 0}).
1
3> ets:update_counter(Tid, test, 1, {test, 0}).
2
```

Reply:
I added the `{read_concurrency, true}` to the ets table. Do you think that it still is an issue? If so, would using `update_counter` be better than adding `{write_concurrency, true}` ?

Reply:
Nope. There's still a race condition there because you're doing 2 separate operations with the table; 1. A lookup, 2. A write. Another process can theoretically write to the table in between 1&2 that invalidates 1. That would not happen with `update_counter` as it is guaranteed to be atomic.